### PR TITLE
Add funcs for getting stored source

### PR DESCRIPTION
**Pull request by JoranAngevaare**
**What is the problem / what does the code in this PR do**
Sometimes you want to process `event_info` but don't know if you have all the required datatypes that are needed for making that data. Rather than doing a try for except, you might from the start want to know if that data can be computed, and where you need to start processing from.

In this PR we add just that.

**Can you briefly describe how it works?**
Look at the target you want to compute, if it's not stored, check if the dependencies are stored. If finally you reach a plugin that is either not stored or not depending on anything (`raw_records`), return `None`, indicating that you cannot make this data.

### CODE REVIEW

File: strax/context.py [hunk 1]
```diff
@@ -1656,6 +1655,52 @@ def copy_to_frontend(self,
...
+        if self.is_stored(run_id, target):
+            return {target}
+
+        deps = strax.to_str_tuple(self._plugin_class_registry[target].depends_on)
+        if not deps:
+            return None
+
+        forbidden = strax.to_str_tuple(self.context_config['forbid_creation_of'])
+        if check_forbidden and target in forbidden:
+            return None
```
Comment (WenzDaniel) [inline strax/context.py:1683]: Maybe be more explicit in this case and raise an error or make a warning. Like you have further down.

**Review by WenzDaniel — APPROVED** [review]
Nice and helpful feature thanks a lot. Just one minor comment.

File: strax/context.py [hunk 2]
```diff
@@ -1656,6 +1655,52 @@ def copy_to_frontend(self,
...
+
+        stored_sources = set()
+        for dep in deps:
+            if self.is_stored(run_id, dep):
```
Comment (jmosbacher) [inline strax/context.py:1687]: isnt this redundent? shouldnt the recursive call to self.get_source check if its stored or not?

**Review by jmosbacher — APPROVED** [review]
Very nice idea, this will probably be used a lot. Just left one small comment (may be possible to make the function a bit smaller==simpler) but even that is not very important.

File: strax/context.py (hunk 2 above)
Comment (JoranAngevaare) [reply strax/context.py]: Ah great point!

File: strax/context.py (hunk 1 above)
Comment (JoranAngevaare) [reply strax/context.py]: good point!

**Pull request merged**